app/src/main/java/org/schabi/newpipe/settings/MainSettingsFragment.java

```diff
@@ -12,16 +15,58 @@
 public class MainSettingsFragment extends BasePreferenceFragment {
     public static final boolean DEBUG = MainActivity.DEBUG;
 
+    private SettingsActivity settingsActivity;
+
     @Override
     public void onCreatePreferences(final Bundle savedInstanceState, final String rootKey) {
-        addPreferencesFromResource(R.xml.main_settings);
+        addPreferencesFromResourceRegistry();
+
+        setHasOptionsMenu(true); // Otherwise onCreateOptionsMenu is not called
 
+        // Check if the app is updatable
         if (!CheckForNewAppVersion.isReleaseApk(App.getApp())) {
-            final Preference update
-                    = findPreference(getString(R.string.update_pref_screen_key));
-            getPreferenceScreen().removePreference(update);
+            getPreferenceScreen().removePreference(
+                    findPreference(getString(R.string.update_pref_screen_key)));
 
             defaultPreferences.edit().putBoolean(getString(R.string.update_app_key), false).apply();
         }
+
+        // Hide debug preferences in RELEASE build variant
+        if (!DEBUG) {
+            getPreferenceScreen().removePreference(
+                    findPreference(getString(R.string.debug_pref_screen_key)));
+        }
```

Comment:
Mmmh, why did you unify the debug and the release `main_settings.xml`? Wasn't it better before, where all classes and resources related to the debug settings were not included at all in the release variant? This also prevented the need to use reflection (I think), since Leak Canary was always available on debug builds, while on release builds the relevant classes are not even compiled so there were no problems.

Reply:
> Mmmh, why did you unify the debug and the release main_settings.xml?

Because we don't have to maintain everything twice (in the XML file).

> Wasn't it better before, where all classes and resources related to the debug settings were not included at all in the release variant?

They were included in the release build before:
![grafik](https://user-images.githubusercontent.com/40789489/148549866-70cc3765-06d9-4b51-80f2-daf353fa67b6.png)